### DISCUSSION TITLE
[Feature] Round trip photo editing

### DISCUSSION THREAD

**Discussion by Clancy508** [immich-app/immich#6592] [Feature Request]
### The feature

At the moment there's lots of options for self hosted photo managers, but as far as I can tell *none* of them support what I would consider core functionality for a photo manager intended to run on a server - round trip photo editing. I know that there's loads of people out there with massive backlogs of RAW photos and such that need sorting and editing, but so far the only suggestion to fix this seems to be to include a basic photo editor in Immich - which is undesirable because it adds a lot of complexity and will never compete with dedicated photo editors.

Instead, what I propose is a workflow for client side photo editing. This would consist of the ability to expose a set of photos in a batch via WebDAV to a client where the user could edit that photos with their choice of image editor, and on dismount Immich would update any edited files, and import any new files including relevant sidecars and edited versions of the images.

I expect this would be easier to implement than a built in image editor, and while it would still be very challenging to do well to me it would be the killer feature - as things stand there isn't a viable option for people who want to edit images in their library without having to do a load of manual organisation (and avoiding that manual work is the entire point of a photo manager).

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by aisbergde** [comment]
This works very well with external libraries: you can edit the content outside immich, immich recognizes changes and updates the data. Unfortunately, it does not work with the standard library, as this assumes that changes can only be made by immich.

**Reply by Clancy508** [reply]
This sounds good but requires sacrificing a lot of the utility of Immich since it can no longer be used as a primary photo organiser (the entire point of external libraries is ceding that organisational control either to manual efforts by the user or to some other system), plus you need to BYO remote access to the library. A built in WebDAV (or any other well supported protocol, WebDAV is just a protocol commonly used for this type of secondary file access function eg in Nextcloud) server allowing access to the primary library would be more integrated into the system by allowing Immich to be the primary organisation system while giving clear signals to the server as to when an image might have been modified/added to, since that access is initiated by the server exposing the data for editing.

**Comment by sharmay** [comment]
I came across https://github.com/igorski/bitmappery/ and wondering if immich can add support photo editing feature using https://github.com/igorski/bitmappery/

I saw this https://github.com/igorski/bitmappery/issues/31 and added a comment.

Feature would be more like a "EDIT" button which can call configure URL and pass info to editing SW, which can edit the photo and upload the edit image.
